examples/tf/trpo_cartpole_recurrent_with_model.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+"""
+This is an example to train a task with TRPO algorithm.
+
+It uses an LSTM-based recurrent policy.
+
+Here it runs CartPole-v1 environment with 100 iterations.
+
+Results:
+    AverageReturn: 100
+    RiseTime: itr 13
+"""
+from garage.experiment import LocalRunner, run_experiment
+from garage.np.baselines import LinearFeatureBaseline
+from garage.tf.algos import TRPO
+from garage.tf.envs import TfEnv
+from garage.tf.optimizers import ConjugateGradientOptimizer
+from garage.tf.optimizers import FiniteDifferenceHvp
+from garage.tf.policies import CategoricalLSTMPolicyWithModel
+
+
+def run_task(*_):
+    """Run task."""
+    with LocalRunner() as runner:
+        env = TfEnv(env_name='CartPole-v1')
+
+        policy = CategoricalLSTMPolicyWithModel(
+            name='policy', env_spec=env.spec)
+
+        baseline = LinearFeatureBaseline(env_spec=env.spec)
+
+        algo = TRPO(
+            env_spec=env.spec,
+            policy=policy,
+            baseline=baseline,
+            max_path_length=100,
+            discount=0.99,
+            max_kl_step=0.01,
+            optimizer=ConjugateGradientOptimizer,
+            optimizer_args=dict(
+                hvp_approach=FiniteDifferenceHvp(base_eps=1e-5)))
+
+        runner.setup(algo, env)
+        runner.train(n_epochs=100, batch_size=4000)
```

Comment:
this is a lot of iters. how long does this test take?

Reply:
Settings are the same as the existing `trpo_cartpole_recurrent`.`trpo_cartpole_recurrent` takes  ~400s while `trpo_cartpole_recurrent_with_model` takes ~460s.

Reply:
oops i thought this was a test. it is concerning that the model version takes 10% longer. i wonder why, but we don't need to worry much about it now.